### ISSUE TITLE
add StopPropagation on click

### ISSUE THREAD

**Issue by ymongo** [ghostdevv/svelte-hamburgers#104]
Hi there, 

This hamburger miss a little modification to be perfect: add stopPropagation on the button: 
```
 <button
     class="hamburger hamburger--{type}"
     aria-label="Hamburger menu"
     class:is-active={open}
     on:click
     on:click|stopPropagation={() => (open = !open)}>
     <span class="hamburger-box">
         <span class="hamburger-inner" />
     </span>
 </button>
```

I copied the Hamburger.svelte and added this mod so it can be used with a click outside lib to close modals or menus (i use [svelte-put/clickoutside](https://github.com/vnphanquang/svelte-put/tree/main/packages/actions/clickoutside)) 

Thanks for the good job

**Comment by ghostdevv** [comment]
I think it would be a breaking change, so next version perhaps :pray:

**Comment by escapingSamsara** [comment]
i thought about this too, def needs close on click feature, but i just worked around by using on:click={() => (open = false)} on the div containing my navlist (which opens after hamburger is clicked), and also added open=false to my handleButtonFnc :)

**Comment by ghostdevv** [comment]
Released in [v5.0.0](https://github.com/ghostdevv/svelte-hamburgers/releases/tag/v5.0.0)
